modules/services/colima.nix

```
@@ -0,0 +1,267 @@
+                '';
+              };
+
+              logFile = lib.mkOption {
+                type = lib.types.path;
+                default = "${config.home.homeDirectory}/.local/state/colima-${name}.log";
+                defaultText = lib.literalExpression "\${config.home.homeDirectory}/.local/state/colima-\${name}.log";
+                description = "Combined stdout and stderr log file for the Colima service.";
+              };
+
+              settings = lib.mkOption {
+                type = yamlFormat.type;
+                default = { };
+                description = "Colima configuration settings, see <https://github.com/abiosoft/colima/blob/main/embedded/defaults/colima.yaml> or run `colima template`.";
+                example = lib.literalExpression ''
+                  {
+                    cpu = 2;
+                    disk = 100;
+                    memory = 2;
+                    arch = "host";
+                    runtime = "docker";
+                    hostname = null;
+                    kubernetes = {
+                      enabled = false;
+                      version = "v1.33.3+k3s1";
+                      k3sArgs = [ "--disable=traefik" ];
+                      port = 0;
+                    };
+                    autoActivate = true;
+                    network = {
+                      address = false;
+                      mode = "shared";
+                      interface = "en0";
+                      preferredRoute = false;
+                      dns = [ ];
+                      dnsHosts = {
+                        "host.docker.internal" = "host.lima.internal";
+                      };
+                      hostAddresses = false;
+                    };
+                    forwardAgent = false;
+                    docker = { };
+                    vmType = "qemu";
+                    portForwarder = "ssh";
+                    rosetta = false;
+                    binfmt = true;
+                    nestedVirtualization = false;
+                    mountType = "sshfs";
+                    mountInotify = false;
+                    cpuType = "host";
+                    provision = [ ];
+                    sshConfig = true;
+                    sshPort = 0;
+                    mounts = [ ];
+                    diskImage = "";
+                    rootDisk = 20;
+                    env = { };
+                  }
+                '';
+              };
+            };
+          }
+        )
+      );
+    };
+  };
+
+  config = lib.mkIf cfg.enable (
+    let
+      activeProfile = lib.findFirst (p: p.isActive) null (lib.attrValues cfg.profiles);
+    in
+    {
+      home.packages = lib.mkIf (cfg.package != null) [ cfg.package ];
+
+      assertions = [
+        {
+          assertion = (lib.count (p: p.isActive) (lib.attrValues cfg.profiles)) <= 1;
+          message = "Only one Colima profile can be active at a time.";
+        }
+      ];
```

```suggestion
      assertions = [
        {
          assertion = (lib.count (p: p.isActive) (lib.attrValues cfg.profiles)) <= 1;
          message = "Only one Colima profile can be active at a time.";
        }
      ];
      
      home.packages = lib.mkIf (cfg.package != null) [ cfg.package ];
```

Not a documented requirement, but i did a recent treewide to keep modules structured in a similar manner.